hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/SourceFormatAdapter.java

```diff
@@ -97,6 +101,7 @@ public InputBatch<JavaRDD<GenericRecord>> fetchNewDataInAvroFormat(Option<String
       case JSON: {
         //sanitizing is done inside the convertor if enabled
         InputBatch<JavaRDD<String>> r = ((Source<JavaRDD<String>>) source).fetchNext(lastCkptStr, sourceLimit);
+        MercifulJsonConverter.clearCache(r.getSchemaProvider().getSourceSchema().getFullName());
```

Comment:
don't we need to clear cache for the other method? fetchNewDataInRowFormat

Reply:
In fecthNewDataInRowFormat, if sanitizing is enabled, it will call fetchNewDataInAvroFormat so that will hit this and clear the cache. Otherwise it isn't sanitizing so the map isn't used